README.md

```diff
@@ -144,30 +133,30 @@ After the client is created, you can start sending events to your Datadog Event
 
 After the client is created, you can start sending Service Checks to Datadog. See the dedicated [Service Check Submission: DogStatsD documentation](https://docs.datadoghq.com/developers/service_checks/dogstatsd_service_checks_submission/?tab=ruby) to see how to submit a Service Check to Datadog.
 
-### Maximum packets size in high-throughput scenarios
+### Maximum packet size in high-throughput scenarios
 
 In order to have the most efficient use of this library in high-throughput scenarios,
-default values for the maximum packets size have already been set for both UDS (8192 bytes)
-and UDP (1432 bytes) in order to have the best usage of the underlying network.
-However, if you perfectly know your network and you know that a different value for the maximum packets
-size should be used, you can set it with the parameter `buffer_max_payload_size`. Example:
+recommended values for the maximum packet size have already been set for both UDS (8192 bytes)
+and UDP (1432 bytes).
+
+However, if are in control of your network and want to use a different value for the maximum packet
+size, you can do it by setting the `buffer_max_payload_size` parameter:
 
 ```ruby
-# Create a DogStatsD client instance.
 statsd = Datadog::Statsd.new('localhost', 8125, buffer_max_payload_size: 4096)
+# ...
+statsd.close()
 ```
 
 ## Threading model
 
-On versions greater than 5.0, we changed the threading model of the library so that one instance of `Datadog::Statsd` could be shared between threads and so that the writes in the socket are non blocking.
+Starting with version 5.0, dogstatsd-ruby employs a new threading model where one instance of `Datadog::Statsd` can be shared between threads and where data sending is non-blocking (asynchronous).
 
-When you instantiate a `Datadog::Statsd`, a companion thread is spawned. This thread will be called the Sender thread, as it is modeled by the [Sender](../lib/datadog/statsd/sender.rb) class. Please use `single_thread: true` while creating an instance if you don't want to or can't use a companion thread.
+When you instantiate a `Datadog::Statsd`, a companion thread is spawned. This thread will be called the Sender thread, as it is modeled by the [Sender](../lib/datadog/statsd/sender.rb) class. You can make use of `single_thread: true` to disable this behavior.
 
-This thread is stopped when you close the statsd client (`Datadog::Statsd#close`). It also means that allocating a lot of statsd clients without closing them properly when not used anymore
-could lead to a thread leak (even though they will be sleeping, blocked on IO).
-The communication between the current thread is managed through a standard Ruby Queue.
+This thread is stopped when you close the statsd client (`Datadog::Statsd#close`). Allocating a lot of statsd clients without calling `#close` after they are not needed anymore can lead to threads being leaked.
```

Comment:
```suggestion
This thread is stopped when you close the statsd client (`Datadog::Statsd#close`). Instantiation a lot of statsd clients without calling `#close` after they are not needed anymore will most likely lead to threads being leaked.
```

Reply:
Should this be "Instantiating a" or Instantiation of"?

Reply:
The former - good catch!

Reply:
I'll add the of in the commit removing L78